### ISSUE TITLE
ha_gateway resource forced to replace after terraform update

### ISSUE THREAD

**Issue by mrizalw** [terraform-google-modules/terraform-google-vpn#99]
### TL;DR

after terraform core update, vpn_ha module wants to replace existing ha_gateway resource

### Expected behavior

ha_gateway resource should not be replace

### Observed behavior

new mandatory variable on the resource made the module plans to replace ha_gateway resource

# module.vpn_ha_nonprod.google_compute_ha_vpn_gateway.ha_gateway[0] must be replaced
  -/+ resource "google_compute_ha_vpn_gateway" "ha_gateway" {
        ~ id         = "projects/xlcoid-nonprod-26e3/regions/asia-southeast2/vpnGateways/vpn-ha-nonprod" -> (known after apply)
          name       = "vpn-ha-nonprod"
        ~ self_link  = "https://www.googleapis.com/compute/v1/projects/xlcoid-nonprod-26e3/regions/asia-southeast2/vpnGateways/vpn-ha-nonprod" -> (known after apply)
        + stack_type = "IPV4_ONLY" # forces replacement
          # (3 unchanged attributes hidden)
  
        ~ vpn_interfaces {
            ~ id                      = 0 -> (known after apply)
            + interconnect_attachment = (known after apply)
            ~ ip_address              = "34.101.23.105" -> (known after apply)
          }
        - vpn_interfaces {
            - id         = 1 -> null
            - ip_address = "34.101.27.57" -> null
          }
      }

### Terraform Configuration

```hcl
vpn.tf

module "vpn_ha_nonprod" {
  source = "terraform-google-modules/vpn/google//modules/vpn_ha"
  #version = "~> 1.3.0"
  project_id       = local.project_id
  region           = local.region_jkt
  network          = local.dev_vpc_selflink
  name             = "vpn-ha-nonprod"
  peer_gcp_gateway = local.prod_vpnha_selflink
  router_asn       = 65002
  router_advertise_config = {
    mode   = "CUSTOM"
    groups = []
    ip_ranges = {
    }
  }
  tunnels = {
    to-gcp-prod-01 = {
      bgp_peer = {
        address = "169.254.1.2"
        asn     = 65001
      }
      bgp_peer_options = {
        advertise_groups = []
        advertise_ip_ranges = {
          <hidden>
        }
        advertise_mode = "CUSTOM"
        route_priority = 999
      }
      bgp_session_range               = "169.254.1.1/30"
      ike_version                     = 2
      vpn_gateway_interface           = 0
      peer_external_gateway_interface = null
      shared_secret                   = "Uop2OP7qWs19ZSgYCFGODKVDkQ5ScW05"
    }
    to-gcp-prod-02 = {
      bgp_peer = {
        address = "169.254.2.2"
        asn     = 65001
      }
      bgp_peer_options = {
        advertise_groups = []
        advertise_ip_ranges = {
          <hidden>
        }
        advertise_mode = "CUSTOM"
        route_priority = 999
      }
      bgp_session_range               = "169.254.2.1/30"
      ike_version                     = 2
      vpn_gateway_interface           = 1
      peer_external_gateway_interface = null
      shared_secret                   = ""
    }
  }
}
```

### Terraform Version

```sh
Terraform v1.3.4
on linux_amd64
```

**Comment by imrannayer** [comment]
@mrizalw can you plz create a bug for Terraform Provider Google [here](https://github.com/hashicorp/terraform-provider-google/issues)?
